Em fix#974

emersonlebleu opened the pull request:
Fix for issue #974

tonydisera commented on client/app/components/pages/GeneHome.vue:
```diff
@@ -2,252 +2,6 @@
  * GeneHome.vue
  *
  */
-<style lang="sass">
```
Would you mind backing out the change that moves styling to the bottom? Then I will be able to see the modified code to introduce the PedigreeGenotypeViz dialog.

emersonlebleu replied:
The only thing I changed here on gene home was moving the styling and taking out that extra v-dialog. I had thought about putting the popup as a home component but ended up deciding to put it in the variantInspectCard. I moved them back to the top but not sure it really helped.

tonydisera commented on client/app/components/viz/VariantInspectCard.vue:
```diff
@@ -967,6 +1052,27 @@ export default {
+        this.showPedigreePopup = true;
+      }
+    },
+    setNumOfSiblings() {
```
I try to avoid setting additional variables in a Vue component unless it is absolutely necessary. The fewer variables, the easier it is to understand the code and the less likely the component is to break with additional changes. How about a condition like this?
`v-if="Object.keys(pedigreeData).length > 5"`

emersonlebleu replied:
This change is completed removed the siblings function and data attribute and replaced with this logic. Appears to render as expected.

tonydisera commented on client/app/components/viz/VariantInspectCard.vue:
```diff
@@ -1248,6 +1352,16 @@ export default {
       }
+      
+      if (self.$refs.popupPedigreeGenotypeViz) {
+        self.$refs.popupPedigreeGenotypeViz.update();
```
I am wondering why we have to call this update() method here. Is there a way to structure the code so any reactivity is just built in automatically?

emersonlebleu replied:
Checked for both of the pedigree charts (popup and regular) and it seems to render just fine without this code. I had to change my popup logic just a little to get it to render without this call but it is working. 

The only thing to consider would be if you expect the updates in the pedigreeGenotypeData to be nested down in the object... the vue documentation says that this can be tricky and you might need a "deep" watcher for that. But if you are okay taking out this code we can it is commented out for now let me know.

tonydisera commented on client/app/components/viz/VariantInspectCard.vue:
```diff
@@ -1248,6 +1352,16 @@ export default {
+      if (self.$refs.popupPedigreeGenotypeViz) {
+        self.$refs.popupPedigreeGenotypeViz.update();
+      } else {
+        setTimeout(function() {
```
We hit this else condition when the $refs variable is not yet initialized. But waiting 2 seconds doesn't guarantee that the variable will be initialized. I recommend removing the code in the else block.

emersonlebleu replied:
Okay lets talk about these comments tomorrow. This particular block is the same as the existing block above it which already was there. Maybe we refactor that part of this more generally. I would agree with your idea of trying to see if we can get these charts to be more "naturally" reactive.

tonydisera reviewed (changes requested):
I am going to give detailed comments and suggestions on these first few pull requests. This will probably be the easiest way to discuss coding style, with a special focus on writing code for reactive components. You will see that I have quite a bit of code that doesn't follow these my own recommendations! ha. We had to learn the hard way when we ported over to Vue.js.

You did an excellent job modifying the code for this issue! Excellent work.

tonydisera commented:
I pulled down this branch and tested this new feature. It works great! I have one small request. Can you move up the title on the popup?  
<img width="994" alt="Screenshot 2023-08-18 at 11 29 08 AM" src="https://github.com/iobio/gene.iobio/assets/8420103/d53d0e9a-6274-43f8-a3a7-5308261d6162">

emersonlebleu commented:
> I pulled down this branch and tested this new feature. It works great! I have one small request. Can you move up the title on the popup? <img alt="Screenshot 2023-08-18 at 11 29 08 AM" width="994" src="https://user-images.githubusercontent.com/8420103/261674482-d53d0e9a-6274-43f8-a3a7-5308261d6162.png">

I moved it slightly higher and changed the structure a tiny bit which makes it pretty easy to move the X button up or down let me know if that looks better? the change is really subtle.
